src/modules/ModuleEIP712.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.25;
+
+// A base for modules that use EIP-712 structured data signing.
+//
+// Unlike other EIP712 libraries, this mixin uses the salt field to hold the account address.
+//
+// It also includes the name, but not a version, in the EIP-712 domain, as modules are intended to be deployed as
+// immutable singletons, thus a different version would have a different module address in the salt.
+//
+// Due to depending on the account address to calculate the domain separator, this abstract contract does not
+// implement EIP-5267, as the domain retrieval function does not provide a parameter to use for the account address
+// (internally the verifyingContract), and the computed `msg.sender` for an `eth_call` without an override is
+// address(0).
+abstract contract ModuleEIP712 {
+    // keccak256(
+    //     "EIP712Domain(uint256 chainId,address verifyingContract,bytes32 salt)"
+    // )
+    bytes32 private constant _DOMAIN_SEPARATOR_TYPEHASH =
+        0x71062c282d40422f744945d587dbf4ecfd4f9cfad1d35d62c944373009d96162;
+
+    function _domainSeparator(address account) internal view returns (bytes32) {
```

Comment:
We will need public view func to get the EIP712Domain fields.

Reply:
I discussed this in the PR body and in comments in the file - we can't safely expose that function from ERC-5267 because the domain separator changes by account, and the function does not take a parameter. We could do something non-standard by depending on `msg.sender` and reverting if `msg.sender == address(0)`, but that is not generally compatible with tools for fetching via `eip712Domain()`.

Reply:
hmm, yeah, we will also need to install `eip712Domain()` as a exe function on the account.

With what we have, we also are not completely compliant with EIP-712.

Reply:
> With what we have, we also are not completely compliant with EIP-712.

How so? I think we're compliant with EIP-712, just not the extension EIP-5267.

Reply:
Look at the [definition of doamin-separator](https://eips.ethereum.org/EIPS/eip-712#definition-of-domainseparator).
>The EIP712Domain fields should be the order as above, skipping any absent fields. Future field additions must be in alphabetical order and come after the above fields. User-agents should accept fields in any order as specified by the EIPT712Domain type.

Reply:
Could you clarify which part is not following the spec? We're using the `chainId`, `verifyingContract`, and `salt` fields, and omitting the `name` and `version`.

Reply:
nvm, we are. Was not thinking the account converted to bytes32 would be the salt.

Actually, we might still be compliant with `ERC5267`.
What if we just return `bytes32(0)` for salt, but have fields to be `11111` to indicate salt is used and dynamic?

Reply:
I think this is a cool idea, but I wonder if doing so requires implementers to keep track of this specific behaviour-- which is functionally equivalent to having a new function. 

I could be wrong but the tradeoff is either "use the old function, but be ready to handle a new special case" or "use a new function." Does that sound right?

Reply:
Right now, we dont have a funciton. One would need to read the code and do custom compute.

The tradeoff you (Peter) mentioned is correct if we want to have a function. I also think it is valuable to add one.
which one is better? I am less sure.

Reply:
> have fields to be 11111 to indicate salt is used and dynamic?

I think the 5267 function `eip712Domain()`'s `fields` value only indicates that it is used in the domain calculation, not whether or not it is dynamic. So if we return 1s for the `verifyingContract`, `chainId`, and `salt`, then the caller would assume that the actual values we return in the view func are used to calculate the domain separator, not that they should be dynamically computed from elsewhere.

So, basically I don't think we can be fully compliant with 5267, because the caller will still always need module-specific context, at which point the caller can just use the known EIP712 domain rather than retrieving it via an `eth_call`.

Reply:
Yep, 5267 compatibility is tricky if routing happens offchain.. Not sure if 5267 compatibility is necessary though, I think its more of a nice to have as the SDK is being used to generate signatures. Since the SDK should know which module it's creating a 1271 signature for, it could call eip712Domain or an equivalent on the module when preparing the signature